### ISSUE TITLE
Restrict dragging  range

### ISSUE THREAD

**Issue by hawkfish** [clauderic/react-sortable-hoc#509]
Nice work!

Not sure if this is possible, but I'd like be able to restrict the drag of an object to a range defined by the object. My use case is a series of operations which have some dependencies. I'd like the user to be able to reorder operations, but not go past dependency boundaries. Is there any way to do this with the current APIs? It doesn't seem like the callbacks can return any information, so all I think I can do right now is put up cursors.

**Comment by clauderic** [comment]
Assuming I'm understanding correctly, have you looked at the `lockToContainerEdges` and `lockOffset` props?

**Comment by hawkfish** [comment]
Yes, but my understanding is that they are not sufficient. `lockToContainerEdges` only restricts the drag t the interior of the entire container, not an element-specific subset. `lockOffset` further restricts the drag of all objects within the container, but does not help with element-specific restrictions.

Basically, the problem is that all the properties defined are global and have to be specified at container rendering time, but the element-specific restriction is not known until the drag starts.

**Comment by danantal** [comment]
I also have this exact same problem.. are there any suggestions on how to solve this?
In my case I am using a sortable list that is also scrollable... and `react-virtualized` on top. And I need to be able to limit the drag area until a specific element, but that might not be in the view if the container is scrolled.
